layers/+misc/all-the-icons/packages.el

```diff
@@ -0,0 +1,135 @@
+;;; packages.el --- all-the-icons layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2018 Sylvain Benner & Contributors
+;;
+;; Author:  <Steven Allen <steven@stebalien.com>>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Commentary:
+;;; Code:
+
+
+(defconst all-the-icons-packages
+  '(
+    (all-the-icons :protected t)
+    ivy-rich
+    (all-the-icons-ivy :toggle (configuration-layer/package-used-p 'ivy))
+    (all-the-icons-ibuffer :toggle (configuration-layer/package-used-p 'ibuffer))
+    (all-the-icons-dired :toggle (configuration-layer/package-used-p 'dired))
+    company-box)
+  "The list of Lisp packages required by the all-the-icons layer.")
+
+(defun all-the-icons/init-all-the-icons ()
+  (use-package all-the-icons
+    :defer t
+    :config
+    ;; EXWM
+    (add-to-list 'all-the-icons-mode-icon-alist
+                 '(exwm-mode all-the-icons-octicon "browser" :face all-the-icons-dsilver))
+    ;; Notmuch
```

Comment:
Should these go in the notmuch layer?

Reply:
Hmmm, I would say no everything all-the-icons specific should be centralised here. However I would be interested how your layer works if for example notmuch is not installed. It should definitely test for the installation status of these packages before trying to modify its config.

Reply:
It shouldn't be a problem if the mode isn't installed. The list contains a tons of non-default modes out of the box anyways.